src/processor/ValueSetProcessor.ts

```diff
@@ -14,14 +15,64 @@ export class ValueSetProcessor {
     }
   }
 
+  static extractRules(input: ProcessableValueSet, target: ExportableValueSet): void {
+    if (input.compose) {
+      input.compose.include?.forEach((vsComponent: any) =>
+        ValueSetProcessor.extractValueSetComponentRules(vsComponent, target, true)
+      );
+      input.compose.exclude?.forEach((vsComponent: any) =>
+        ValueSetProcessor.extractValueSetComponentRules(vsComponent, target, false)
+      );
+    }
+  }
+
+  static extractValueSetComponentRules(
+    vsComponent: any,
+    target: ExportableValueSet,
+    include: boolean
+  ) {
+    if (ValueSetProcessor.isValueSetComponent(vsComponent)) {
+      target.rules.push(ValueSetConceptComponentRuleExtractor.process(vsComponent, include));
+    }
+    target.rules = compact(target.rules);
```

Comment:
If we make this function `private` then I think we can safely remove the `compact` here and call it at the end of `extractRules` instead.  Otherwise we're calling it in multiple places once we merge in the care rule extractor.

Reply:
That's a good point. No reason to have this happen in the individual extractors when it can safely happen at the end.